core/src/utils/test/press-keys.ts

```diff
@@ -0,0 +1,140 @@
+  page: Page;
+  context: BrowserContext;
+
+  constructor({ page }: { page: Page }) {
+    this.page = page;
+    this.context = page.context();
+    this.browser = this.context.browser()!;
+  }
+
+  pressKeys: typeof pressKeys = pressKeys.bind(this);
+}
+
+const baseModifiers = {
+  primary: (_isApple: any) => (_isApple() ? [COMMAND] : [CTRL]),
+  primaryShift: (_isApple: any) => (_isApple() ? [SHIFT, COMMAND] : [CTRL, SHIFT]),
+  primaryAlt: (_isApple: any) => (_isApple() ? [ALT, COMMAND] : [CTRL, ALT]),
+  secondary: (_isApple: any) => (_isApple() ? [SHIFT, ALT, COMMAND] : [CTRL, SHIFT, ALT]),
+  access: (_isApple: any) => (_isApple() ? [CTRL, ALT] : [SHIFT, ALT]),
+  ctrl: () => [CTRL],
+  alt: () => [ALT],
+  ctrlShift: () => [CTRL, SHIFT],
+  shift: () => [SHIFT],
+  shiftAlt: () => [SHIFT, ALT],
+  undefined: () => [],
+};
+
+const isAppleOS = () => process.platform === 'darwin';
+const isWebkit = (page: Page) => page.context().browser()!.browserType().name() === 'webkit';
+const browserCache = new WeakMap();
+
+/**
+ * Detects if the browser has natural tab navigation.
+ * Natural tab navigation means that the browser will focus the next element
+ * in the DOM when the tab key is pressed.
+ */
+const getHasNaturalTabNavigation = async (page: Page) => {
+  if (!isAppleOS() || !isWebkit(page)) {
+    return true;
+  }
+  if (browserCache.has(page.context().browser()!)) {
+    return browserCache.get(page.context().browser()!);
+  }
+  const testPage = await page.context().newPage();
+  await testPage.setContent(`<button>1</button><button>2</button>`);
+  await testPage.getByText('1').focus();
+  await testPage.keyboard.press('Tab');
+  const featureDetected = await testPage.getByText('2').evaluate((node) => node === document.activeElement);
+  browserCache.set(page.context().browser()!, featureDetected);
+  await testPage.close();
+  return featureDetected;
+};
+
+type Options = {
+  /**
+   * Number of times to press the key.
+   */
+  times?: number;
+  /**
+   * Delay between each key press in milliseconds.
+   */
+  delay?: number;
+};
+
+const modifiers = {
+  ...baseModifiers,
+  shiftAlt: (_isApple: () => boolean) => (_isApple() ? [SHIFT, ALT] : [SHIFT, CTRL]),
+};
+
+/**
+ * Presses a key combination.
+ * @param key - Key combination to press.
+ * @param options - Options for the key press.
+ * @example
+ * ```ts
+ * await pressKeys('a');
+ * await pressKeys('a', { times: 2 });
+ * await pressKeys('a', { delay: 100 });
+ * await pressKeys('Shift+Tab');
+ * ```
+ */
+export async function pressKeys(this: PageUtils, key: string, { times, ...pressOptions }: Options = {}) {
+  const hasNaturalTabNavigation = await getHasNaturalTabNavigation(this.page);
+  const keys = key.split('+').flatMap((keyCode) => {
+    if (Object.prototype.hasOwnProperty.call(modifiers, keyCode)) {
+      return modifiers[keyCode as keyof typeof modifiers](isAppleOS).map((modifier) =>
+        modifier === CTRL ? 'Control' : capitalCase(modifier)
+      );
```

Comment:
Can we document what this is doing?

Reply:
Let me know if the updated documentation makes sense 👍